fc/permission_request.go

```diff
@@ -0,0 +1,244 @@
+package fullcontact
+
+//Permission Request
+
+type PermissionRequestOption func(ar *PermissionRequest)
+
+type PermissionRequest struct {
+	Timestamp  			int							`json:"timestamp,omitempty"`
+	Query    			*MultifieldRequest			`json:"query,omitempty"`
+	ConsentPurposes		[]*ConsentPurpose 			`json:"consentPurposes,omitempty"`
+	Locale     			string      				`json:"locale,omitempty"`
+	IpAddress			string						`json:"ipAddress,omitempty"`
+	Language     		string      				`json:"language,omitempty"`
+	CollectionMethod	string						`json:"collectionMethod,omitempty"`
+	CollectionLocation 	string      				`json:"collectionLocation,omitempty"`
+	Tcf					string						`json:"tcf,omitempty"`
+	PolicyUrl     		string      				`json:"policyUrl,omitempty"`
+	TermsService		string						`json:"termsService,omitempty"`
+	PurposeId			int							`json:"purposeId,omitempty"`
+	Channel				string						`json:"channel,omitempty"`
+}
+
+func NewPermissionRequest(option ...PermissionRequestOption) (*PermissionRequest, error) {
+	permissionRequest := &PermissionRequest{}
+
+	for _, opt := range option {
+		print(option)
+		opt(permissionRequest)
+	}
+	return permissionRequest, nil
+}
+
+func validateForPermissionCreateFields(request *PermissionRequest) error {
+	if request.ConsentPurposes == nil {
+		return NewFullContactError("At least 1 `consentPurpose` is Required for PermissionRequest")
+	} else if !isPopulated(request.CollectionMethod) {
+		return NewFullContactError("Collection Method is required for PermissionRequest")
+	} else if !isPopulated(request.CollectionLocation) {
+		return NewFullContactError("Collection Location is required for PermissionRequest")
+	} else if !isPopulated(request.PolicyUrl) {
+		return NewFullContactError("Policy URL is required for PermissionRequest")
+	} else if !isPopulated(request.TermsService) {
+		return NewFullContactError("Terms of Service is required for PermissionRequest")
+	}
+	return nil
+}
+
+func validateForPermissionVerifyFields(request *PermissionRequest) error {
+	if request.PurposeId == 0 {
+		return NewFullContactError("Purpose ID is required for PermissionRequest")
+	} else if !isPopulated(request.Channel) {
+		return NewFullContactError("Channel is required for PermissionRequest")
+	}
+	return nil
+}
+
+func validateForPermissionCreate(request *PermissionRequest) error {
+	err := validateForPermissionCreateFields(request)
+	if err != nil {
+		return err
+	}
+	for _, consentPurpose := range request.ConsentPurposes{
+		print(consentPurpose)
```

Comment:
There is a print statement. Can you search in the whole project for any other missed

Reply:
Removed print